packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts

```diff
@@ -188,7 +190,13 @@ export class AmplifyGraphqlApi extends Construct {
         ...defaultTranslationBehavior,
         ...(translationBehavior ?? {}),
       },
-    });
+    };
+
+    if (isSqlModelDataSourceBinding(definition.modelDataSourceBinding)) {
+      this.extendTransformConfig(executeTransformConfig, definition.modelDataSourceBinding);
+    }
```

Comment:
This appears to be a side-effecting function, but also returns the updated object. Let's adhere to one or the other. I think longer-term going to a 'transformConfigBuilder' may make this more readable, but I'm fine keep that OOS for this change.

Reply:
D'oh! Thanks, I intended to move away from the side effect in favor of immutable objects, and didn't close the task. Will fix.

Reply:
Resolved in bd11853f3